### PR TITLE
Feature product category filtering and channel product identifier

### PR DESCRIPTION
**Pull request by gevashkar-rampersadh**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,8 +42,8 @@
     <swagger-maven-plugin-version>3.1.3</swagger-maven-plugin-version>
     <logback-version>1.0.13</logback-version>
     <major-version>5</major-version>
-    <minor-version>20</minor-version>
-    <patch-version>0</patch-version>
+    <minor-version>21</minor-version>
+    <patch-version>0-SNAPSHOT</patch-version>
```
Comment (Shaaheen) [inline pom.xml:46]: final

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 This page describes changes to the Airtime Service Interface implemented across different releases of the interface.
 
+## v5.21.0
+Released TBD 2020
```
Comment (Shaaheen) [inline src/docs/devguide/hugo/site/content/release-notes.md:4]: date

File: src/main/java/io/electrum/airtime/api/MsisdnResource.java [hunk 3]
```diff
@@ -108,39 +109,74 @@ public final void lookupMsisdn(
...
-           @Context Request request,
-           @Suspended AsyncResponse asyncResponse,
-           @Context HttpHeaders httpHeaders,
-           @Context UriInfo uriInfo,
-           @Context HttpServletRequest httpServletRequest) {
+         @ApiParam(value = "The Msisdn. This must conform to the ITU E.164 numbering plan (https://www.itu.int/rec/T-REC-E.164/en).", required = true) @QueryParam(LookupMsisdn.QueryParameters.MSISDN) @Pattern(regexp = "^\\+?[1-9]\\d{1,14}") @NotNull String msisdn,
+         @ApiParam(value = "The provider who processed the original purchase attempt.") @QueryParam(LookupMsisdn.QueryParameters.OPERATOR) String operator,
+         @ApiParam(value = "Used to filter the products to lookup for a given msisdn.") @QueryParam(LookupMsisdn.QueryParameters.PRODUCT_TYPE) String productType,
+         @ApiParam(value = "Used to filter the products to lookup for a given channel.") @QueryParam(LookupMsisdn.QueryParameters.CHANNEL_NAME) String channelName,
+         @ApiParam(value = "Used to filter the products to lookup within a specific category.") @QueryParam(LookupMsisdn.QueryParameters.PRODUCT_CATEGORY) String productCategory,
```
Comment (Shaaheen) [inline src/main/java/io/electrum/airtime/api/MsisdnResource.java:162]: I'm a bit worried that people will be confused on how to use and differentiate the productType and productCategory fields. Is there any examples we could put in the docs to make a bit clearer what a potential product category is? Or maybe we could make the productTypes docs better and say that a value in there must be set to either  [AIRTIME_FIXED, AIRTIME_VARIABLE, SMS_BUNDLE, DATA, APP_RELATED, COMBO, DATA_FIXED, DATA_VARIABLE, VOICE_BUNDLE]. Or maybe you could link to the Product object and say that it has to conform to that type field:  https://electrumpayments.github.io/airtime-service-interface-docs/specification/definitions/#product (so we don't have to manually update this in future)

**Review by Shaaheen — CHANGES_REQUESTED** [review]

File: src/main/java/io/electrum/airtime/api/MsisdnResource.java (hunk 3 above)
Comment (gevashkar-rampersadh) [reply src/main/java/io/electrum/airtime/api/MsisdnResource.java:162]: I think definitely pointing at the product type field is the better choice, however linking to the actual URL will be a problem is we ever change how the API is hosted, so maybe we can just point to the product type.

**Review by Shaaheen — APPROVED** [review]
LGTM (Just the date in the release notes pls)

**Pull request merged**